ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_8/TransactionServiceVisibilityIT.scala

```diff
@@ -141,6 +141,110 @@ class TransactionServiceVisibilityIT extends LedgerTestSuite {
       }
   })
 
+  test(
+    "TXTreeChildOrder",
+    "Trees formed by childEventIds should be maintaining Transaction event order",
+    allocate(TwoParties, SingleParty, SingleParty),
+    enabled = _.committerEventLog.eventLogType.isCentralized,
+    tags = privacyHappyCase(
+      asset = "Transaction Tree",
+      happyCase = "Trees formed by childEventIds should be maintaining Transaction event order",
+    ),
+  )(implicit ec => {
+    case Participants(
+          Participant(alpha, alice, gbp_bank),
+          Participant(beta, bob),
+          Participant(delta, dkk_bank),
+        ) =>
+      for {
+        gbpIouIssue <- alpha.create(gbp_bank, Iou(gbp_bank, gbp_bank, "GBP", 100, Nil))
+        gbpTransfer <-
+          alpha.exerciseAndGetContract(gbp_bank, gbpIouIssue.exerciseIou_Transfer(_, alice))
+        dkkIouIssue <- delta.create(dkk_bank, Iou(dkk_bank, dkk_bank, "DKK", 110, Nil))
+        dkkTransfer <-
+          delta.exerciseAndGetContract(dkk_bank, dkkIouIssue.exerciseIou_Transfer(_, bob))
+
+        aliceIou1 <- eventually("exerciseIouTransfer_Accept") {
+          alpha.exerciseAndGetContract(alice, gbpTransfer.exerciseIouTransfer_Accept(_))
+        }
+        aliceIou <- eventually("exerciseIou_AddObserver") {
+          alpha.exerciseAndGetContract(alice, aliceIou1.exerciseIou_AddObserver(_, bob))
+        }
+        bobIou <- eventually("exerciseIouTransfer_Accept") {
+          beta.exerciseAndGetContract(bob, dkkTransfer.exerciseIouTransfer_Accept(_))
+        }
+
+        trade <- eventually("create") {
+          alpha.create(
+            alice,
+            IouTrade(alice, bob, aliceIou, gbp_bank, "GBP", 100, dkk_bank, "DKK", 110),
+          )
+        }
+        tree <- eventually("exerciseIouTrade_Accept") {
+          beta.exercise(bob, trade.exerciseIouTrade_Accept(_, bobIou))
+        }
+
+        aliceTree <- eventually("transactionTreeById1") {
+          alpha.transactionTreeById(tree.transactionId, alice)
+        }
+        bobTree <- beta.transactionTreeById(tree.transactionId, bob)
+        gbpTree <- eventually("transactionTreeById2") {
+          alpha.transactionTreeById(tree.transactionId, gbp_bank)
+        }
+        dkkTree <- eventually("transactionTreeById3") {
+          delta.transactionTreeById(tree.transactionId, dkk_bank)
+        }
+        aliceTrees <- alpha.transactionTrees(alice)
```

Comment:
We test here both Transaction Tree and Transaction Tree By Id, since they boil down to different API usage.